repondeur/zam_repondeur/views/lectures.py

```diff
@@ -0,0 +1,185 @@
+class LecturesAdd:
+    def __init__(self, request: Request) -> None:
+        self.request = request
+        self.data_dir = self.request.registry.settings["zam.data_dir"]
+
+    @view_config(request_method="GET")
+    def get(self) -> dict:
+        return self._form_data()
+
+    @view_config(request_method="POST")
+    def post(self) -> Response:
+        chambre = self.request.POST["chambre"]
+        session = self.request.POST["session"]
+        num_texte = self.request.POST["num_texte"]
+        if chambre not in CHAMBRES:
+            raise HTTPBadRequest
+
+        lecture_dir = os.path.join(self.data_dir, f"{chambre}-{session}-{num_texte}")
+        if os.path.isdir(lecture_dir):
+            self.request.session.flash(("warning", "Cette lecture existe déjà..."))
+        else:
+            os.makedirs(lecture_dir)
+            self.request.session.flash(("success", "Lecture créée avec succès."))
+
+        return HTTPFound(
+            location=self.request.route_url(
+                "lecture", chambre=chambre, session=session, num_texte=num_texte
+            )
+        )
+
+    def _form_data(self) -> dict:
+        return {
+            "chambres": CHAMBRES.items(),
+            "sessions": [(sess, sess) for sess in SESSIONS],
+        }
+
+
+@view_config(route_name="lecture", renderer="lecture.html")
+def lecture(request: Request) -> dict:
+    return {
+        "lecture": Lecture(  # type: ignore
+            chambre=request.matchdict["chambre"],
+            session=request.matchdict["session"],
+            num_texte=request.matchdict["num_texte"],
+        )
+    }
+
+
+@view_config(route_name="amendements_csv")
+def amendements_csv(request: Request) -> Response:
+    chambre = request.matchdict["chambre"]
+    session = request.matchdict["session"]
+    num_texte = request.matchdict["num_texte"]
+    assert chambre in CHAMBRES
+
+    try:
+        amendements = get_amendements_senat(session, num_texte)
+    except NotFound:
+        request.session.flash(("danger", "Aucun amendement n'a pu être trouvé."))
+        return HTTPFound(
+            location=request.route_url(
+                "lecture", chambre=chambre, session=session, num_texte=num_texte
+            )
+        )
+
+    with NamedTemporaryFile() as file_:
+
+        tmp_file_path = os.path.abspath(file_.name)
+
+        write_csv(amendements, tmp_file_path)
+
+        response = FileResponse(tmp_file_path)
+        attach_name = f"amendements-{chambre}-{session}-{num_texte}.csv"
+        response.content_type = "text/csv"
+        response.headers["Content-Disposition"] = f"attachment; filename={attach_name}"
+        return response
+
+
+@view_config(route_name="amendements_xlsx")
+def amendements_xlsx(request: Request) -> Response:
+    chambre = request.matchdict["chambre"]
+    session = request.matchdict["session"]
+    num_texte = request.matchdict["num_texte"]
+    assert chambre in CHAMBRES
```

Comment:
I'm a bit concerned about using assert on something which might be optimized, hence set [`__debug__` to False](https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement), do you have any experience with that?

Reply:
That's a good point.

My idea was that we would add some proper validation in the near future, and the `assert` was a cheap sanity check in the meantime.

Reply:
Fixed in 9c35ef6